tasks/Module 4/task_1.py

```diff
@@ -0,0 +1,27 @@
+"""Mad libs."""
+import os
+import re
+
+TEMPLATE = ['ADJECTIVE', 'NOUN', 'ADVERB', 'VERB']
+
+
+def mad_lib(source_file_path):
+    """Repace placeholders with user inputs."""
+    tmp_file = source_file_path + '.tmp'
+    with open(source_file_path, 'r') as read_file:
+        with open(tmp_file, 'w') as write_file:
+            for line in read_file:
+                pattern = re.compile(f'{"|".join(TEMPLATE)}')
```

Comment:
This pattern does not change between lines of read_file, so you can move it outside the loop. Actually you can even make a module level constant from it: `PLACEHOLDERS_PATTERN`.